onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc

```diff
@@ -121,7 +121,7 @@ Status ResizeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   std::vector<int32_t> sizes_hw;
   std::vector<int32_t> axes;
   const bool is_nhwc = model_builder.GetPreferredLayout() == DataLayout::NHWC;
-  if (input_defs.size() == 3) {  // Use scales.
+  if (input_defs.size() == 3 || !input_defs[2]->Name().empty()) {  // Use scales.
```

Comment:
I think this would read more clearly if you used `>= 2` rather than `==3`, which makes it appear like you're interesting in index 3, but actually you care about index 2 (index 2 is the scales). Plus, it's more future proof if more tensors are added later to use `>=` over `==`. Likewise elsewhere.

Reply:
This is simplified by `GetTensorName`.